src/Compiler/CompilerUtils.cpp

```diff
@@ -251,6 +237,20 @@ struct Command {
 // =============================================================================
 // Methods for compiling and file processing.
 
+std::string getOnnxMlirFullVersion() {
+  const std::string OnnxMlirVersion = "onnx-mlir version 0.3.0";
+  return
+#ifdef ONNX_MLIR_VENDOR
```

Comment:
This has gotten fairly complicated at this point. I would consider having defaults for everything in order to simplify the logic and be able to test it in all cases.

Reply:
By default, `getOnnxMlirFullVersion()` returns `onnx-mlir version 0.3.0`.
If `ONNX_MLIR_VENDOR`, then it returns `ONNX_MLIR_VENDOR, onnx-mlir version 0.3.0`.
If not `ONNX_MLIR_VENDOR` and both repo and revision are defined, then it returns e.g., `onnx-mlir version 0.3.0 (git@github.com:onnx/onnx-mlir.git fd6a145f7702f3a1789cd6923890efd55f67cf64, git@github.com:llvm/llvm-project.git 9778ec057cf4214241e4a970f3e764e3cf150181)`.
This output logic is similar to LLVM. 

The logic actually doesn't change in this PR. Do you want to elaborate on your idea about default values? 
If we have a default value for `ONNX_MLIR_VENDOR`, then we will never generate repo and revision.
If we first check if `ONNX_MLIR_VENDOR` is the default value, then the logic is not simplified.
Similarly, if we have default values for repo and revision, then we don't want to output them if they are the default value, unless we want to output e.g., `onnx-mlir version 0.3.0 (unknown onnx-mlir repo, unknown onnx-mlir revision, unknown llvm repo, unknown llvm revision)`, which is first different from LLVM, and doesn't look good.

Reply:
The purpose of a version is for the users to report it for identifying the code involved. Yes a repo and revision would certainly identify the code but these are "internal" matter which the users don't care about and are hard to remember. A short released version number is typically tagged to a commit hash so it WILL uniquely identify the code and at the same time is more user friendly. That's why you typically never see repo and revision reported in the version for released code.

Reply:
What I see is that this function was simply moved from line 156. The only functional change is the removal of the storage specifier `static`, which is to allow other compilation units to call this function. So looks fine from my perspective.

Reply:
I have created a new file Version.cpp to make it more similar to LLVM implementation, and I have added link references to LLVM implementations as PR comments.